### ISSUE TITLE
Doesn't work with HTTPS

### ISSUE THREAD

**Issue by adrianmihalko** [icflorescu/iisexpress-proxy#9]
Dear developer,

I have a.NET app, which works only in localhost, because if I am trying to connect to it remotely I get a Bad Request error. I am trying to redirect localhost port to my public interface:

C:\Users\a>iisexpress-proxy https://localhost:41959 to 41000
IIS Express Proxy 1.1.8
Proxying https://localhost:41959 to network interfaces:
        Local Area Connection: 192.168.1.10:41000
Listening... [press Control-C to exit]

...and the result is:

**https**://192.168.1.10:41959: can't open the page

a$ curl https://192.168.1.10:3000 -v
- Rebuilt URL to: https://192.168.1.10:3000/
-   Trying 192.168.1.10...
- Connected to 192.168.1.10 (192.168.1.10) port 3000 (#0)
- WARNING: using IP address, SNI is being disabled by the OS.
- Server aborted the SSL handshake
- Closing connection 0
  curl: (35) Server aborted the SSL handshake

**http**://192.168.1.10:41959:

Aw snap! Something went wrong. Check your console to see the error.

Log:
Error: unable to verify the first certificate
    at Error (native)
    at TLSSocket.<anonymous> (_tls_wrap.js:1060:38)
    at emitNone (events.js:86:13)
    at TLSSocket.emit (events.js:185:7)
    at TLSSocket._finishInit (_tls_wrap.js:584:8)
    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:416:38)
Listening... [press Control-C to exit]

I think because we redirect https to http. Is there any way to keep the https connection? My app can listen on https only.

Something like this:

C:\Users\a>iisexpress-proxy https://localhost:41959 to https://:41000

Regards,
Adrian

**Comment by GarageWeb** [comment]
I have a similar issue, I have to use https for development and it does not work.

**Comment by icflorescu** [comment]
Sorry for not being able to address this. At the moment I'm in a long awaited holiday with sporadic Internet connectivity and no other device than a mobile phone.
Feel free to play around with the code, and if you manage to come up with a PR, I'll pull it.

**Comment by adrianmihalko** [comment]
Here is a fixed version, it works for me.

Changes are around protocolPrefix, port + proxyPort is set in the code.

[iisproxy-fixed.txt](https://github.com/icflorescu/iisexpress-proxy/files/467073/iisproxy-fixed.txt)

**Comment by sharpmachine** [comment]
@adrianmihalko this worked for me. Thanks. Anyway you could create a PR for @icflorescu? It would be super helpful for everyone else using this awesome module as well.

**Comment by icflorescu** [comment]
I'm more than willing to accept a PR! :-)

**Comment by kristijan-ujevic** [comment]
Howdy, first of all thanks for the tool it helps a lot :)

You should really update the code with the above fix.

**Comment by digital-synapse** [comment]
On my machine ssl connections  are failing through as well. Was the above fix ever submitted as a pull request?

**Comment by icflorescu** [comment]
@digital-synapse – thank you for the PR! I've pulled it and will publish a new version on `npm` in a couple of hours.

As stated above, I'm more than happy to accept PRs with useful features. I seldom use `iisexpress-proxy` myself nowadays and never expected it to gain so much attention and usage when I decided to publish it, years ago. This is an open-source project, and the essence of open-source is not only **free to use**, but also **a moral obligation to contribute** for the benefit of the community. So, to all users benefitting from it, if you think a specific functionality should be included in the project, by all means, do a PR!...

Raising an issue helps, but coming up with the actual code helps a lot more :-)

Thanks again!
